apps/prairielearn/src/pages/instructorAssessmentQuestions/instructorAssessmentQuestions.html.ts

```diff
@@ -0,0 +1,399 @@
+            <div class="modal-dialog" role="document">
+              <div class="modal-content">
+                <div class="modal-header">
+                  <h4 class="modal-title" id="resetQuestionVariantsModalLabel">
+                    Confirm reset question variants
+                  </h4>
+                </div>
+                <div class="modal-body">
+                  <p>
+                    Are your sure you want to reset all current variants of this question?
+                    <strong>All ungraded attempts will be lost.</strong>
+                  </p>
+                  <p>Students will receive a new variant the next time they view this question.</p>
+                </div>
+                <div class="modal-footer">
+                  <form name="reset-question-variants-form" method="POST">
+                    <input type="hidden" name="__action" value="reset_question_variants" />
+                    <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+                    <input
+                      type="hidden"
+                      name="unsafe_assessment_question_id"
+                      class="js-assessment-question-id"
+                      value=""
+                    />
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">
+                      Cancel
+                    </button>
+                    <button type="submit" class="btn btn-danger">Reset question variants</button>
+                  </form>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          ${renderEjs(
+            __filename,
+            "<%- include('../partials/assessmentSyncErrorsAndWarnings'); %>",
+            resLocals,
+          )}
+
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white d-flex align-items-center">
+              ${resLocals.assessment_set.name} ${resLocals.assessment.number}: Questions
+            </div>
+            ${AssessmentQuestionsTable({
+              questions,
+              assessmentType: resLocals.assessment.type,
+              urlPrefix: resLocals.urlPrefix,
+              hasCourseInstancePermissionEdit:
+                resLocals.authz_data.has_course_instance_permission_edit,
+            })}
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function AssessmentQuestionsTable({
+  questions,
+  urlPrefix,
+  assessmentType,
+  hasCourseInstancePermissionEdit,
+}: {
+  questions: AssessmentQuestionRow[];
+  assessmentType: string;
+  urlPrefix: string;
+  hasCourseInstancePermissionEdit: boolean;
+}) {
+  let nTableCols = 11;
+
+  // If at least one question has a nonzero unlock score, display the Advance Score column
+  const showAdvanceScorePercCol =
+    questions.filter((q) => {
+      return q.assessment_question_advance_score_perc !== 0;
+    }).length >= 1;
+  if (showAdvanceScorePercCol) nTableCols++;
```

Comment:
Nit, we can condense this a bit (this will need some Prettier action):

```suggestion
  // If at least one question has a nonzero unlock score, display the Advance Score column
  const showAdvanceScorePercCol =
    questions.filter((q) => q.assessment_question_advance_score_perc !== 0)
    .length >= 1;
    
  let nTableCols = showAdvanceScorePercCol ? 12 : 11;
```